### PR TITLE
Add training script

### PR DESCRIPTION
**Pull request by bhavitvyamalik**

### CODE REVIEW

**Comment by bhavitvyamalik** [comment]
@gchhablani BLEU score has been added along with Tensorboard logging. Let me know if any changes are required

File: data/translate_cc12m_MBart50.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    "es_XX": p_generate_es_XX,
+    "de_DE": p_generate_de_DE,
+    # "ru_RU": p_generate_ru_RU,
+}
+
+def run_generate(input_str, p_generate):
+    inputs = tokenizer(input_str, return_tensors="jax", padding="max_length", truncation=True, max_length=64)
+    p_inputs = shard(inputs.data)
+    output_ids = p_generate(p_params, p_inputs)
+    output_strings = tokenizer.batch_decode(output_ids.reshape(-1, 64), skip_special_tokens=True, max_length=64)
```
Comment (patrickvonplaten) [inline data/translate_cc12m_MBart50.py:81]: No need to do the `.reshape(...)` here I think

File: main.py [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+    state = jax_utils.unreplicate(state)
+    logger.info(f"Saving Checkpoint in {save_dir}")
+    ckpt_save_dir = f"{save_dir}/ckpt-{mb_item(state.step)-1}"
+    if os.path.exists(ckpt_save_dir) and not overwrite:
+        logger.info("checkpoint exists, skipping overwrite")
+    else:
+        model.save_pretrained(
+            ckpt_save_dir,
+            params=state.params,
+            push_to_hub=False,
```
Comment (patrickvonplaten) [inline main.py:308]: think you can just do `push_to_hub=push_to_hub` here instead of the special "push_to_hub" code below

File: main.py [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+        # masks = batch.pop("attention_mask")
+        logits = model(batch["pixel_values"], batch["input_ids"], batch["attention_mask"], params=params, train=False)[0]
+        loss = loss_fn(logits, labels, batch["attention_mask"], label_smoothing_factor)
+
+        # summarize metrics
+        metrics = {"loss": loss}
+        metrics = jax.lax.pmean(metrics, axis_name="batch")
+        return metrics
+
+    num_beams = 4  # model has beam size 5, should we keep 4 or 5 here?
```
Comment (patrickvonplaten) [inline main.py:701]: 4 is definitely good enough - maybe even reducing it to 2 makes sense here

File: main.py [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+    gen_kwargs = {"max_length": data_args.max_seq_length, "num_beams": num_beams}
+
+    def generate_step(params, batch):
+        model.params = params
+        output_ids = model.generate(batch["pixel_values"], **gen_kwargs)  # forced_bos_token_id, decoder_start_token_id, or bos_token_id
```
Comment (patrickvonplaten) [inline main.py:706]: I would pass the params here instead `model.generate(...., params=params)`

File: main.py [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+        # summarize metrics
+        metrics = {"loss": loss}
+        metrics = jax.lax.pmean(metrics, axis_name="batch")
+        return metrics
+
+    num_beams = 4  # model has beam size 5, should we keep 4 or 5 here?
+    gen_kwargs = {"max_length": data_args.max_seq_length, "num_beams": num_beams}
+
+    def generate_step(params, batch):
+        model.params = params
```
Comment (patrickvonplaten) [inline main.py:705]: I'd deleet this

**Pull request merged**